src/Grids/convenience_constructors.jl

```diff
@@ -0,0 +1,366 @@
+`ExtrudedFiniteDifferenceGrid` with a
+`FiniteDifferenceGrid` vertical grid and a
+`SpectralElementGrid1D` horizontal grid.
+ - ``
+"""
+SliceXZGrid(; kwargs...) = SliceXZGrid(Float64; kwargs...)
+function SliceXZGrid(
+    ::Type{FT};
+    z_elem::Integer,
+    x_min::Real,
+    x_max::Real,
+    z_min::Real,
+    z_max::Real,
+    x1periodic::Bool,
+    Nq::Integer,
+    n1::Integer,
+    device::ClimaComms.AbstractDevice = ClimaComms.device(),
+    context::ClimaComms.AbstractCommsContext = ClimaComms.context(device),
+    stretch::Meshes.StretchingRule = Meshes.Uniform(),
+    hypsography::HypsographyAdaption = Flat(),
+    global_geometry::Geometry.AbstractGlobalGeometry = Geometry.CartesianGlobalGeometry(),
+    quad::Quadratures.QuadratureStyle = Quadratures.GLL{Nq}(),
+) where {FT}
+    check_device_context(context, device)
+
+    x1boundary = (:east, :west)
+    z_boundary_names = (:bottom, :top)
+    h_domain = Domains.IntervalDomain(
+        Geometry.XPoint{FT}(x_min),
+        Geometry.XPoint{FT}(x_max);
+        periodic = x1periodic,
+        boundary_names = x1boundary,
+    )
+    h_mesh = Meshes.IntervalMesh(h_domain; nelems = n1)
+    h_topology = Topologies.IntervalTopology(context, h_mesh)
+    h_grid = Grids.SpectralElementGrid1D(h_topology, quad)
+    z_domain = Domains.IntervalDomain(
+        Geometry.ZPoint{FT}(z_min),
+        Geometry.ZPoint{FT}(z_max);
+        boundary_names = z_boundary_names,
+    )
+    z_mesh = Meshes.IntervalMesh(z_domain, stretch; nelems = z_elem)
+    z_topology = Topologies.IntervalTopology(context, z_mesh)
+    vertical_grid = FiniteDifferenceGrid(z_topology)
+    return ExtrudedFiniteDifferenceGrid(
+        h_grid,
+        vertical_grid,
+        hypsography,
+        global_geometry,
+    )
+end
+
+"""
+    RectangleXYGrid(
+        ::Type{<:AbstractFloat}; # defaults to Float64
+        x_min::Real,
+        x_max::Real,
+        y_min::Real,
+        y_max::Real,
+        x1periodic::Bool,
+        x2periodic::Bool,
+        Nq::Integer,
+        n1::Integer, # number of horizontal elements
+        n2::Integer, # number of horizontal elements
+        device::ClimaComms.AbstractDevice = ClimaComms.device(),
+        context::ClimaComms.AbstractCommsContext = ClimaComms.context(device),
+        hypsography::HypsographyAdaption = Flat(),
+        global_geometry::Geometry.AbstractGlobalGeometry = Geometry.CartesianGlobalGeometry(),
+        quad::Quadratures.QuadratureStyle = Quadratures.GLL{Nq}(),
+    )
+
+A convenience constructor, which builds a
+`SpectralElementGrid2D` with a horizontal
+`RectilinearMesh` mesh.
+"""
+RectangleXYGrid(; kwargs...) = RectangleXYGrid(Float64; kwargs...)
+function RectangleXYGrid(
+    ::Type{FT};
+    x_min::Real,
+    x_max::Real,
+    y_min::Real,
+    y_max::Real,
+    x1periodic::Bool,
+    x2periodic::Bool,
+    Nq::Integer,
+    n1::Integer, # number of horizontal elements
+    n2::Integer, # number of horizontal elements
```

Comment:
```suggestion
    x_elem::Integer, # number of horizontal elements
    y_elem::Integer, # number of horizontal elements
```
so that it follows `z_elem`?

Reply:
Done!